unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PlayerPassportHUDController.cs

```diff
@@ -85,13 +96,40 @@ private void OnCurrentPlayerIdChanged(string current, string previous)
             }
             else
             {
+                QueryNftCollections(currentUserProfile.userId);
                 userProfileBridge.RequestFullUserProfile(currentUserProfile.userId);
                 currentUserProfile.OnUpdate += UpdateUserProfile;
                 view.SetPassportPanelVisibility(true);
                 UpdateUserProfileInSubpanels(currentUserProfile);
             }
         }
 
+        private void QueryNftCollections(string userId)
+        {
+            if (string.IsNullOrEmpty(userId))
+                return;
+
+            Environment.i.platform.serviceProviders.theGraph.QueryNftCollections(userId, NftCollectionsLayer.ETHEREUM)
+                       .Then((nfts) => ownedNftCollectionsL1 = nfts)
+                       .Catch((error) => Debug.LogError(error));
+
+            Environment.i.platform.serviceProviders.theGraph.QueryNftCollections(userId, NftCollectionsLayer.MATIC)
+                       .Then((nfts) => ownedNftCollectionsL2 = nfts)
+                       .Catch((error) => Debug.LogError(error));
```

Comment:
I think it would be a good idea to encapsulate all the functions related to "requesting data to the backend" in a separated class (for example `WebInterfacePassportApiBridge`) that you could inject in each controller that need it. It is something that we are already doing with friends (`WebInterfaceFriendsApiBridge`) and it make the code clearer. What do you think?